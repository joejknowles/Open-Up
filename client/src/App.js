import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    const response = 'date';
    return (
      <div className="App">
        Today { response }
      </div>
    );
  }
}

export default App;
