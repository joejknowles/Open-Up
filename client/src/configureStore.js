import topReducer from './reducers';
import { createStore, applyMiddleware } from 'redux'
