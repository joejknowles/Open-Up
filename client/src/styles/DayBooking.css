.Heading {
  width: 45%;
  margin: auto;
  position: relative;
}

.next-day {
  position: absolute;
  right: 0;
  top: 0;
}

.prev-day {
  position: absolute;
  left: 0;
  top: 0;
}

.List {
  width: 45%;
  margin: auto;
}

.Slot {
  width: 100%;
  height: 4em;
  line-height: 4em;
  margin: auto;

  margin-bottom: 1em;
}

.unavailableSlot {
  border-style: solid;
  border-width: 2px;
  border-color: #BBB;
  border-radius: 0.5em;
  color: #BBB;
}

.message {
  display: inline-block;
  vertical-align: middle;
  line-height: normal;
}

.BookButton {
  font-size: 16px;
  width: 100%;
  height: 100%;
  background-color: white;
  border-color: black;
  border-radius: 0.5em;
  outline: none;
}

.BookButton:focus {
  -webkit-box-shadow: 6px 6px 30px -8px rgba(0,0,0,0.75);
  -moz-box-shadow: 6px 6px 30px -8px rgba(0,0,0,0.75);
  box-shadow: 6px 6px 30px -8px rgba(0,0,0,0.75);
}

.BookButton:active {
  -webkit-box-shadow: inset 6px 6px 30px -8px rgba(0,0,0,0.75);
  -moz-box-shadow: inset 6px 6px 30px -8px rgba(0,0,0,0.75);
  box-shadow: inset 6px 6px 30px -8px rgba(0,0,0,0.75);
}
