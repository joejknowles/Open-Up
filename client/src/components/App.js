import React, { Component } from 'react';
import '../styles/App.css';
import DayBooking from './dayBooking';
import Errors from './alerts/errors';

import { Provider } from 'react-redux';

class App extends Component {
  render() {
    return (
      <Provider store={ this.props.store }>
        <div className="App">
          <Errors />
          <DayBooking />
        </div>
      </Provider>
    );
  }
}

export default App;
