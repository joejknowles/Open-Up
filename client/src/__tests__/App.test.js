import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import { shallow } from 'enzyme';
import DayBooking from '../components/dayBooking';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);
const emptyStoreMock = mockStore({ allSlots: [] });

it('renders without crashing', () => {
  shallow(<App store={ emptyStoreMock }/>);
});

it('renders booking component', () => {
  const app = shallow(<App store={ emptyStoreMock }/>);
  expect(app.find(DayBooking).length).toBe(1);
});

it('renders correctly', () => {
  const tree = renderer.create(<App store={ emptyStoreMock }/>).toJSON();
  expect(tree).toMatchSnapshot();
});
