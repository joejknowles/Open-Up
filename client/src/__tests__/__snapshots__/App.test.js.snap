exports[`test renders correctly 1`] = `
<div
  className="App">
  <div>
    <h3>
      Today 
      
    </h3>
  </div>
</div>
`;
